src/bindings/c/docs/how_to_wrap_openvino_objects_with_c.md

```diff
@@ -0,0 +1,88 @@
+# How to wrap OpenVINO objects with C
+
+Here is the details about how to wrap objects(C++) form OpenVINO to objects(C).
+
+In OpenVINO C++ implementation, many objects are defined with `class`, such as `ov::Core`, `ov::Model`, `ov::InferRequest`..., but for C `class` doesn't be supported. So, C need to create new object to represent those objects. Three kinds of methods had been adopted in our implementation:
+ * C `struct` contains a shared pointer to C++ `class`, [Wrap by C++ Shared Pointer](#wrap_by_c++_shared_pointer)
+ * C `struct` contains a instance of C++ `class`, [Wrap by C++ Object](#wrap_by_c++_object)
+ * C `struct` rewrite the C++ `class`, [Wrap by Rewrite](#wrap_by_rewrite)
+
+Tips:
+1) For the objects which needs to be hided for users, C `struct` contains a shared pointer will be adopted.
+2) For the objects which needs to be created, operated and read by users, rewrite the C++ `class` will be better.
+3) For some simple objects, C `struct` contains a instance of C++ `class` will be enough.
+
+In OpenVINO C++, most objects implemented with C++ `class` as following:
+
+```
+class ClassName {
```

Comment:
Looks like we shouldn't provide the base principles of C++ language here :) Developer already should know it

Reply:
removed